sonic_platform_base/sonic_storage/ssd.py

```diff
@@ -166,6 +166,20 @@ def parse_generic_ssd_info(self):
             else:
                 temp_raw = temp_raw.split()[-2]
                 self.temperature = float(temp_raw)
+
+            spare_blocks_raw = self._parse_re('Available Spare\s*(.+?)\n', self.ssd_info)
+            if spare_blocks_raw != NOT_AVAILABLE:
+                spare_blocks_raw = spare_blocks_raw.split()[-1]
+                self.reserved_blocks = float(spare_blocks_raw.strip('%'))
+
+            disk_io_reads_raw = self._parse_re('Data Units Read\s*(.+?)\n', self.ssd_info)
+            if disk_io_reads_raw != NOT_AVAILABLE:
+                self.disk_io_reads = disk_io_reads_raw.split(':')[-1].strip()
+
+            disk_io_writes_raw = self._parse_re('Data Units Written\s*(.+?)\n', self.ssd_info)
+            if disk_io_writes_raw != NOT_AVAILABLE:
+                self.disk_io_writes = disk_io_writes_raw.split(':')[-1].strip()
+
         else:
```

Comment:
@assrinivasan would be better to write saparat function for nvme and non-nvme case?

Reply:
Split nvme case into its own function in latest commit